ts/WoltLabSuite/Core/Component/Confirmation.ts

```diff
@@ -78,37 +81,69 @@ class ConfirmationPrefab {
     });
   }
 
-  async softDelete(title?: string, askForReason = false): Promise<ResultConfirmationWithReason> {
+  async softDelete(): Promise<ResultSoftDeleteWithoutReason>;
+  async softDelete(title: string): Promise<ResultSoftDeleteWithoutReason>;
+  async softDelete(title: string, askForReason: false): Promise<ResultSoftDeleteWithoutReason>;
+  async softDelete(title: string, askForReason: true): Promise<ResultConfirmationWithReason>;
+  async softDelete(
+    title?: string,
+    askForReason = false,
+  ): Promise<ResultConfirmationWithReason | ResultSoftDeleteWithoutReason> {
     let question: string;
     if (title === undefined) {
       question = getPhrase("wcf.dialog.confirmation.softDelete.indeterminate");
     } else {
       question = getPhrase("wcf.dialog.confirmation.softDelete", { title });
     }
 
-    return this.withReason(question, askForReason);
+    if (askForReason) {
+      return this.withReason(question, true);
+    }
+
+    const dialog = dialogFactory().withoutContent().asConfirmation();
+    dialog.show(question);
+
+    return new Promise<ResultSoftDeleteWithoutReason>((resolve) => {
+      dialog.addEventListener("primary", () => {
+        resolve({
+          result: true,
+        });
+      });
+
+      dialog.addEventListener("cancel", () => {
+        resolve({
+          result: false,
+        });
+      });
+    });
   }
 
-  async withReason(question: string, askForReason: boolean): Promise<ResultConfirmationWithReason> {
+  async withReason(question: string, isOptional: boolean): Promise<ResultConfirmationWithReason> {
     const dialog = dialogFactory().withoutContent().asConfirmation();
 
     let reason: HTMLTextAreaElement | undefined = undefined;
-    if (askForReason) {
-      const id = DomUtil.getUniqueId();
-      const label = getPhrase("wcf.dialog.confirmation.reason");
+    const id = DomUtil.getUniqueId();
+    const label = getPhrase(isOptional ? "wcf.dialog.confirmation.reason.optional" : "wcf.dialog.confirmation.reason");
```

Comment:
Should this use the asterisk instead?

Reply:
No, the asterisk is a horrible design decision. It should have always been the other way around by marking optional fields with an explicit label.